tests/unit/drivers/test_encodingevaluation_driver.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+from jina.drivers.evaluate import EncodeEvaluationDriver
+from jina.drivers.helper import DocGroundtruthPair, array2pb
+from jina.executors.evaluators.encode import BaseEncodingEvaluator
+from jina.proto import jina_pb2
+
+
+class MockDiffEvaluator(BaseEncodingEvaluator):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def name(self):
+        return 'MockDiffEvaluator'
+
+    def evaluate(self, doc_embedding: 'np.array', groundtruth_embedding: 'np.array', *args, **kwargs) -> float:
+        """"
+        :param doc_embedding: the embedding of the document (resulting from an Encoder)
+        :param groundtruth_embedding: the expected embedding of the document
+        :return the evaluation metric value for the request document
+        """
+        return abs(sum(doc_embedding - groundtruth_embedding) / len(doc_embedding))
+
+
+class SimpleEvaluateDriver(EncodeEvaluationDriver):
+    @property
+    def exec_fn(self):
+        return self._exec_fn
+
+
+def test_encoding_evaluate_driver():
+    def create_document_ground_truth_pairs(num_docs):
+        pairs = []
+        for idx in range(num_docs):
+            doc = jina_pb2.Document()
+            gt = jina_pb2.Document()
+            doc.embedding.CopyFrom(array2pb(np.array([1, 1])))
+            gt.embedding.CopyFrom(array2pb(np.array([2, 2])))
+            pairs.append(DocGroundtruthPair(doc=doc, groundtruth=gt))
+        return pairs
```

Comment:
for example, here you're creating a test to test encoding evaluate driver, while put a nested function inside to create resources, I'd think create a fixture can be more clear (you decouple the resources from test itself). I'm not a big fan of nested functions.

Reply:
In the orher case, do I still need to define the class outside and pass an instance as fixture? or how would it be refactored?